src/main/java/frc/robot/commands/vision/CalculateStdDevs.java

```diff
@@ -0,0 +1,139 @@
+import frc.robot.constants.Constants;
+import frc.robot.subsystems.Drivetrain;
+import frc.robot.util.LogManager;
+import frc.robot.util.Vision;
+
+/**
+ * Calculates standard deviations for vision
+ */
+public class CalculateStdDevs extends CommandBase {
+  private final ShuffleboardTab m_tab;
+  private final Drivetrain m_drive;
+  private final Vision m_vision;
+  private ArrayList<Pose2d> m_poses;
+  private int m_arrayLength;
+  private Timer m_endTimer;
+
+  /**
+   * Constructor for CalculateStdDevs
+   * @param time How long to run the command, also how many poses to use
+   * @param shuffleboardTab The shuffleboard tab to output to
+   * @param drive The drivetrain
+   * @param vision The vision
+   */
+  public CalculateStdDevs(int time, ShuffleboardTab shuffleboardTab, Drivetrain drive, Vision vision) {
+    m_tab = shuffleboardTab;
+    m_drive = drive;
+    m_vision = vision;
+    m_arrayLength = time;
+  }
+
+  /**
+   * Resets the pose array
+   */
+  @Override
+  public void initialize() {
+    m_poses = new ArrayList<Pose2d>();
+  }
+
+  /**
+   * Adds a pose to the array
+   */
+  @Override
+  public void execute() {
+    Pose2d pose = m_vision.getPose2d(m_drive.getPose());
+    // If the pose exists, add it to the first open spot in the array
+    if (pose != null) {
+      m_endTimer.stop();
+      m_endTimer.reset();
+      m_poses.add(pose);
+      System.out.printf("%.1f%% done", ((double)m_poses.size())/m_arrayLength * 100);
+    } else {
+      m_endTimer.start();
+      // If 5 seconds have passed since it saw an April tag, stop the command
+      // Prevents it from running forever
+      if(m_endTimer.hasElapsed(5)){
+        cancel();
+      }
+    }
+  }
+
+  /**
+   * Calculates the standard deviation
+   */
+  @Override
+  public void end(boolean interrupted) {
+    // If the array is empty, don't try to calculate std devs
+    if(m_poses.size() == 0){
+      System.out.println("There are no poses in the array\nTry again where the robot can see an April tag.");
+      return;
+    }
+    
+    // Calculate std devs
+    double meanX = 0;
+    double meanY = 0;
+    double meanRot = 0;
+    for (int i = 0; i < m_poses.size(); i++) {
+      meanX += m_poses.get(i).getX();
+      meanY += m_poses.get(i).getY();
+      meanRot += m_poses.get(i).getRotation().getRadians();
+    }
+    meanX /= m_poses.size();
+    meanY /= m_poses.size();
+    meanRot /= m_poses.size();
+    double totalX = 0;
+    double totalY = 0;
+    double totalRot = 0;
+    for (int i = 0; i < m_poses.size(); i++) {
+      totalX += Math.pow(m_poses.get(i).getX() - meanX, 2);
+      totalY += Math.pow(m_poses.get(i).getY() - meanY, 2);
+      totalRot += Math.pow(m_poses.get(i).getRotation().getRadians() - meanRot, 2);
+    }
+    double stdDevX = Math.sqrt(totalX / m_poses.size());
+    double stdDevY = Math.sqrt(totalY / m_poses.size());
+    double stdDevRot = Math.sqrt(totalRot / m_poses.size());
```

Comment:
it should be : Math.sqrt(total / (m_poses.size()-1))
because it is a sample and not a population

Reply:
I thought it was the square root of the average oft he difference squared. To take the average, it adds up all of the values and divides by the number of values. Why would it be -1 for a sample?